components/google-cloud/google_cloud_pipeline_components/experimental/tensorflow_probability/anomaly_detection/tfp_anomaly_detection.py

```diff
@@ -23,204 +23,106 @@ def tfp_anomaly_detection(input_dataset: Input[Dataset],
-            tf.math.log1p(-anomaly_threshold / 2.) * tf.ones_like(predictive_mean)  # Target log CDF at upper bound.
-        ],
-        axis=0)
-    limits, _, _ = tfp.math.find_root_chandrupatla(
-        objective_fn=lambda x: target_log_cdfs - predictive_dist.log_cdf(x),
-        low=tf.stack([
-            predictive_mean - 100 * predictive_stddev,
-            predictive_mean - 10 * predictive_stddev
-        ],
-                     axis=0),
-        high=tf.stack([
-            predictive_mean + 10 * predictive_stddev,
-            predictive_mean + 100 * predictive_stddev
-        ],
-                      axis=0))
-
-    # Identify anomalies.
-    anomalies = np.less(tail_probabilities, anomaly_threshold)
-    if mask is not None:
-      anomalies = np.logical_and(anomalies, ~mask)
-    observed_anomalies = list(
-        itertools.compress(range(len(times)), list(anomalies)))
-
-    times = times.strftime('%Y-%m-%d %H:%M:%S').tolist()
-    observed_series = observed_series.numpy().tolist()
-    predictive_mean = predictive_mean.numpy().tolist()
-    lower_limit = limits[0].numpy().tolist()
-    upper_limit = limits[1].numpy().tolist()
-    tail_probabilities = tail_probabilities.numpy().tolist()
-
-    return PredictionOutput(times, times, observed_series, predictive_mean,
-                            lower_limit, upper_limit, observed_anomalies,
-                            tail_probabilities)
-
   data = load_data(input_dataset.path)
-  regularized_series = tfp.sts.regularize_series(data)
-
-  model = default_model.build_default_model(regularized_series)
-  surrogate_posterior = tfp.sts.build_factored_surrogate_posterior(
-      model, seed=seed)
-  _ = tfp.vi.fit_surrogate_posterior(
-      target_log_prob_fn=model.joint_log_prob(regularized_series),
-      surrogate_posterior=surrogate_posterior,
-      optimizer=tf.optimizers.Adam(0.1),
-      num_steps=max_num_steps,
-      jit_compile=jit_compile,
-      convergence_criterion=(
-          tfp.optimizer.convergence_criteria.SuccessiveGradientsAreUncorrelated(
-              window_size=20, min_num_steps=50)))
-
-  posterior_samples = surrogate_posterior.sample(num_samples, seed=seed)
-
-  predictions = detect_anomalies(regularized_series, model, posterior_samples,
-                                 anomaly_threshold)
+  logger.info(
+      'Input dataset has {0} rows. If you run out of memory you should increase set_memory_limit in your pipeline.'
+      .format(len(data)))
+  predictions = tfp_ad.detect_anomalies(data, anomaly_threshold,
+                                        use_gibbs_predictive_dist,
+                                        num_warmup_steps, num_samples,
+                                        jit_compile, seed)
   predictions_df = format_predictions(predictions)
   predictions_df.to_csv(output_dataset.path)
 
 
 def generate_component_file():
   packages = [
-      'numpy', 'pandas', 'tf-nightly',
+      'pandas', 'tf-nightly',
```

Comment:
Could you add a comment, comment on tf-nightly is only required while we are waiting for tf v2.6 release.

Reply:
Done.